grpc/src/main/java/com/linecorp/armeria/server/grpc/JsonUnframedGrpcErrorHandler.java

```diff
@@ -0,0 +1,203 @@
+        final String grpcStatusDetailsBin = trailers.get(GrpcHeaderNames.GRPC_STATUS_DETAILS_BIN);
+        final ResponseHeaders responseHeaders = ResponseHeaders.builder(httpStatus)
+                                                               .contentType(MediaType.JSON_UTF_8)
+                                                               .addInt(GrpcHeaderNames.GRPC_STATUS,
+                                                                       grpcCode.value())
+                                                               .build();
+        boolean success = false;
+        try (OutputStream outputStream = new ByteBufOutputStream(buffer);
+             JsonGenerator jsonGenerator = mapper.createGenerator(outputStream)) {
+            jsonGenerator.writeStartObject();
+            jsonGenerator.writeNumberField("code", grpcCode.value());
+            jsonGenerator.writeStringField("grpc-code", grpcCode.name());
+            if (grpcMessage != null) {
+                jsonGenerator.writeStringField("message", grpcMessage);
+            }
+            if (cause != null && ctx.config().verboseResponses()) {
+                jsonGenerator.writeStringField("stack-trace", Exceptions.traceText(cause));
+            }
+            if (!Strings.isNullOrEmpty(grpcStatusDetailsBin)) {
+                com.google.rpc.Status rpcStatus = null;
+                try {
+                    rpcStatus = decodeGrpcStatusDetailsBin(grpcStatusDetailsBin);
+                } catch (InvalidProtocolBufferException e) {
+                    logger.warn("Unexpected exception while decoding grpc-status-details-bin: {}",
+                                grpcStatusDetailsBin, e);
+                }
+                if (rpcStatus != null) {
+                    jsonGenerator.writeFieldName("details");
+                    writeErrorDetails(rpcStatus.getDetailsList(), jsonGenerator);
+                }
+            }
+            jsonGenerator.writeEndObject();
+            jsonGenerator.flush();
+            success = true;
+        } catch (IOException e) {
+            logger.warn("Unexpected exception while generating a JSON response", e);
+        } finally {
+            if (!success) {
+                buffer.release();
+            }
+        }
+        if (success) {
+            return HttpResponse.of(responseHeaders, HttpData.wrap(buffer));
+        } else {
+            return HttpResponse.of(responseHeaders);
+        }
+    }
+
+    @VisibleForTesting
+    void writeErrorDetails(List<Any> details, JsonGenerator jsonGenerator) throws IOException {
+        jsonGenerator.writeStartArray();
+        for (Any detail : details) {
+            try {
+                jsonMarshaller.writeValue(detail, jsonGenerator);
+            } catch (IOException e) {
+                logger.warn("Unexpected exception while writing an error detail to JSON. detail: {}",
+                            detail, e);
```

Comment:
Could we log the `ctx` as well for better debuggability? i.e.

```
    void writeErrorDetails(ServiceRequestContext ctx, List<Any> details, JsonGenerator jsonGenerator) throws IOException {
```

Please note that you can create a `ServiceRequestContext` with `ServiceRequestContext.of()` or `.builder()` in your tests.